src/functions-and-arrays.js

```diff
@@ -1,18 +1,109 @@
 // Iteration #1: Find the maximum
 
+function maxOfTwoNumbers(number1, number2){
+  return Math.max(number1, number2);
+}
+
+
+
 // Iteration #2: Find longest word
+
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+function findLongestWord(words) {
+  let word ='';
+  if (!words.length){
+    return null;
+  } else for (let i = 0; i < words.length; i++) {
+    if (word.length < words[i].length) {
+      word = words[i];
+    }
+  }
+  return word;
+};
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+  function sumNumbers (arr){
+   
+    let sum = 0;
+    for (let i = 0 ; i < arr.length ; i++){
+      sum += arr[i]
+    }
+    return sum
+  }
+
+  // Bonus - Iteration #3.1: A generic sum() function
+  
+  const mixedArr = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
+  function sum(mixedArr){
+    let sum = 0;
+    //console.log("array length",mixedArr.length)
+    if (!mixedArr.length){
+      return 0;
+    }
+    else{
+    for (let i = 0 ; i < mixedArr.length ; i++){
+      if (typeof mixedArr[i] === "number"){
+        console.log(sum)
+      sum += mixedArr[i]
+      //console.log("number",sum)
+      } else if ((typeof mixedArr[i] === "boolean") && (mixedArr[i] = "true")) {
```

Comment:
I see two issues on this line. Firstly, I think you'd like to compare "===" mixedArr[i] with "true" rather than asigning "true" to mixedArr[i]. Secondly, you want to see whether mixArr[i] is _true_. true is a boolean, whereas "true" is a string.